### DISCUSSION TITLE
scalar upload in federation always empty

### DISCUSSION THREAD

**Discussion by harmnot** [99designs/gqlgen#1789] [Q&A]
### What happened?
trying to upload file on my Apollo federation but always got `map[string]interface {} is not an Upload`
anyone has same issue ?? for the gateway, I am following from the example on this repo also, been find those issue how to fix for hours, but finally I ask on here because I am using this library :)

example mutation with Altair 
```
mutation($image: Upload!) {
  createFile(file: $image)
}


// add file on Altair given name image here 
```
### versions
 - `gqlgen version`?  latest
 - `go version`? latest
 - dep or go modules? mod

**Comment by davae1an** [comment]
im getting the same issue with grapql-request

**Comment by pwaterz** [comment]
Same issue here trying to upload a file via apollo client in react. I am able to upload a file via postman though.

Edit:

I had to update my Apollo client setup as follows:

```
import { createUploadLink } from 'apollo-upload-client';

const httpLink = createUploadLink({ uri: graphqlUri });

```

The important part is using createUploadLink instead of the default HttpLink2. Reference where I found the answer https://medium.com/@enespalaz/file-upload-with-graphql-9a4927775ef7
